### PR TITLE
GroceryItem DataBase Design

### LINKED ISSUE #4: Grocery list table and database design
- discuss & decide what tables are needed for the grocery list.
- how will data be stored for the grocery list
- implement the code to reflect on Postgres

### PR DESCRIPTION
**Pull request by chiashenglin0204**
GroceryItem DataBase design
Close #4

### CODE REVIEW

File: src/entities/GroceryItem.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+@Entity('groceryItems')
+export class GroceryItem extends BaseEntity {
+	// ---------------- fields ----------------
+	@Field()
+	@PrimaryGeneratedColumn({ type: 'int' })
+	id!: number;
+
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	groceryItemInfoId!: number; // foreign key to Fridge table
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts]: Maybe we don't need a groceryItemInfo table. We can merge the groceryItemInfo table and the groceryItem table together.

File: src/entities/GroceryItem.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	groceryListId!: number; // foreign key to Fridge table
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts]: fix the comment

File: src/entities/GroceryItem.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	quantity!: number;
+
+	// ---------------- relationship ----------------
+	@ManyToOne(() => GroceryList, groceryList => groceryList.groceryItems)
+	groceryList: GroceryList;
+
+	@ManyToOne(
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts]: if we merge the two tables, we don't need this relationship

File: src/entities/GroceryItem.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	)
+	groceryItemInfo: GroceryItemInfo;
+
+	// ---------------- time ----------------
+	@Field(() => String)
+	@CreateDateColumn({ type: 'timestamptz' })
+	createdAt: Date;
+}
+
+@ObjectType()
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts]: this class is probably not needed if we merge the two tables together

File: src/entities/GroceryItemInfo.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	userId!: number; // foreign key to User table
+
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	measurementTypeId!: number; // foreign key to MeasurementType table
+
+	// ---------------- relationship ----------------
+	@ManyToOne(
```
Comment (hungnguyen3) [inline src/entities/GroceryItemInfo.ts]: not needed due to merging table

File: src/entities/GroceryList.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+	@Field(() => Number)
+	@Column({ type: 'int' })
+	ownerId!: number; // foreign key to users table
+
+	// ---------------- relationship ----------------
+	@ManyToOne(() => User, user => user.groceryLists)
+	owner: User;
+
+	@OneToMany(() => GroceryListUserTable, gLUTable => gLUTable.groceryList)
```
Comment (hungnguyen3) [inline src/entities/GroceryList.ts]: maybe change the 'gLU' to 'glu'

File: src/entities/MeasurementType.ts [hunk 7]
```diff
@@ -30,4 +31,10 @@ export class MeasurementType extends BaseEntity {
 		fridgeItemInfo => fridgeItemInfo.measurementTypeId
 	)
 	fridgeItemInfo: FridgeItemInfo;
+
+	@OneToMany(
```
Comment (hungnguyen3) [inline src/entities/MeasurementType.ts]: we'll probably merge this, so don't forget to change this too

File: src/entities/User.ts [hunk 8]
```diff
@@ -53,6 +56,15 @@ export class User extends BaseEntity {
 	@OneToMany(() => FridgeItemInfo, fridgeItemInfo => fridgeItemInfo.user)
 	fridgeItemInfos: FridgeItemInfo[];
 
+	@OneToMany(() => GroceryList, groceryList => groceryList.owner)
+	groceryLists: GroceryList[];
+
+	@OneToMany(() => GroceryListUserTable, gLUTable => gLUTable.user)
```
Comment (hungnguyen3) [inline src/entities/User.ts]: merging don't forget to delete 1 relationship

**Review by hungnguyen3 — CHANGES_REQUESTED** [review]
good work! Tag me again once changes are made

File: src/entities/MeasurementType.ts (hunk 7 above)
Comment (chiashenglin0204) [reply src/entities/MeasurementType.ts]: Fix my life

File: src/entities/GroceryItem.ts [hunk 9]
```diff
@@ -19,52 +18,38 @@ export class GroceryItem extends BaseEntity {
...
+	@Column({ type: 'int' })
 	userId!: number; // foreign key to User table
 
 	@Field(() => Number)
+	@Column({ type: 'int' })
 	measurementTypeId!: number; // foreign key to MeasurementType table
 
-	@Field(() => String)
-	measurement!: string;
+	// ---------------- relationship ----------------
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts:51]: added relationships for the foreign keys of measurementType & user on the next commit

File: src/entities/GroceryItem.ts [hunk 10]
```diff
@@ -8,7 +8,6 @@ import {
 } from 'typeorm';
 import { ObjectType, Field } from 'type-graphql';
 import { GroceryList } from './GroceryList';
-import { GroceryItemInfo } from './GroceryItemInfo';
 @ObjectType()
 @Entity('groceryItems')
```
Comment (hungnguyen3) [inline src/entities/GroceryItem.ts:15]: added a unique combination of upc & userid on the next commit

**Review by hungnguyen3 — APPROVED** [review]
looks good! Thanks so much for ur contribution ;)

**Pull request merged**